posts/models.py

```diff
@@ -1,13 +1,91 @@
 from django.db import models
 from django.utils import timezone
 from django.contrib.auth.models import User
+from django.core.validators import FileExtensionValidator
+from django.core.validators import MaxValueValidator
+from django.core.validators import MinValueValidator
 
 
 # --------------------------------- Post: parent of Poll And Resume----------------------------------
 
-
 class Post(models.Model):
     post_id = models.AutoField(primary_key=True)
     author = models.ForeignKey(User, on_delete=models.CASCADE)
     description = models.TextField()
     date_posted = models.DateTimeField(default=timezone.now)
+
+
+# ------------------------------------------- Resume-------------------------------------------------
+
+
+class Resume(Post):
+    resume_file = models.FileField(upload_to='files', validators=[FileExtensionValidator(allowed_extensions=['pdf'])])
+
+    def __str__(self):
+        return f"Resume {self.post_id} by {self.author}"
+
+    def get_average_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.get_rating_average()
+        return rating_sum / len(self.rating_set.all())
+
+    def get_design_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.design_rating
+        return rating_sum / len(self.rating_set.all())
+
+    def get_skill_relevance_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.skill_relevance_rating
+        return rating_sum / len(self.rating_set.all())
+
+    def get_grammar_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.grammar_rating
+        return rating_sum / len(self.rating_set.all())
+
+    def get_conciseness_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.conciseness_rating
+        return rating_sum / len(self.rating_set.all())
```

Comment:
I think you can use these methods as [class method](https://stackabuse.com/pythons-classmethod-and-staticmethod-explained/), so you can use [cls.objects.filter](https://github.com/redhat-beyond/falcon/blob/c719a8365b2f146224f763478d978acad5c39c02/tasks/models.py#L37)

Reply:
Ok, I'll check that out, thanks!

Reply:
If I use these classes as class methods I can't use the object it self to get it's rating_set, so I don't see how am I supposed to get the rating average with the method as class method? @natifridman

Reply:
I thought it was in the Rating model, you can ignore this comment if you keep it in the Resume model